build/tasks/GetDocXmlFiles.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace RepoTasks
+{
+    /// <summary>
+    /// Filters a list of .xml files to only those that are .NET Xml docs files
+    /// </summary>
+    public class GetDocXmlFiles : Microsoft.Build.Utilities.Task
+    {
+        [Required]
+        public ITaskItem[] Files { get; set; }
+
+        [Output]
+        public ITaskItem[] XmlDocFiles { get; set; }
+
+        public override bool Execute()
+        {
+            var xmlDocs = new ConcurrentBag<ITaskItem>();
+            Parallel.ForEach(Files, f =>
+            {
+                try
+                {
+                    using (var file = File.OpenRead(f.ItemSpec))
+                    using (var reader = new StreamReader(file))
+                    {
+                        string line;
+                        for (var i = 0; i < 2; i++)
+                        {
+                            line = reader.ReadLine();
```

Comment:
instead of essentially writing your custom XML parser, is there an xml parse we can use to ensure we have a top level <doc> element instead?

Reply:
I tried that. Unfortunately, some xml docsfiles contain corrupted xml, so XDocument.Load fails to parse. :-/. It seems VS and roslyn are more forgiving on XML correctness.

By the way, this detected that _all_ .xml files in the restore folder are xmldocs. I didn't find any .xml files in packages which were not xmldocs. Is that your expectation too?

Reply:
That's what I found as well.

Reply:
FYI if there are corrupted xmls, we should probably at least notify the relevant teams that it's not proper XML, even though it may not matter for VS

Reply:
That's a really good point. Did a quick check...looks like it was only one file out of 3505.
```
/Users/namc/dev/aspnet/Universe21/obj/pkgs/netstandard.library/2.0.3/build/netstandard2.0/ref/netstandard.xml	The 'p' start tag on line 654 position 2 does not match the end tag of 'th'. Line 654, position 86.
```

Reply:
https://github.com/dotnet/standard/issues/794